api4/team.go

```diff
@@ -103,6 +105,24 @@ func getTeamsForUser(c *Context, w http.ResponseWriter, r *http.Request) {
 	}
 }
 
+func getTeamsUnreadForUser(c *Context, w http.ResponseWriter, r *http.Request) {
+	c.RequireUserId()
+	if c.Err != nil {
+		return
+	}
+
+	// optional team id to be excluded from the result
+	teamID := r.URL.Query().Get("id")
```

Comment:
Since this is different from APIv3 and you can specify the user, you need to do a permission check like this:
```
if c.Session.UserId != c.Params.UserId && !app.SessionHasPermissionTo(c.Session, model.PERMISSION_MANAGE_SYSTEM) {
	c.SetPermissionError(model.PERMISSION_MANAGE_SYSTEM)
	return
}
```